gatsby/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,153 @@
+import classNames from 'classnames';
+import React, { useCallback, useState } from 'react';
+import Col from '../col';
+import Container from '../container';
+import GovIcon from '../gov-icon';
+import Row from '../row';
+import PaginationButton from './pagination-button';
+import classes from './pagination.module.scss';
+
+interface IProps {
+  /** How many items should be on one page.
+   * ```25 | 50 | 75 | 100``` are recommended,
+   * however default is **10**.
+   * You can pass any number.
+   */
+  perPage?: 10 | 25 | 50 | 75 | 100 | number;
+  /** Class applied to container */
+  className?: string;
+  /** Class applied to ```<Col>``` for each children */
+  childClassName?: string;
+}
+
+/**
+ * This is a complete mess. I hope this code is so
+ * awful I'm never allowed to write code again.
+ */
+
+/**
+ * Pass any components inside as children.
+ * Component will automatically create pagination for them.
+ * Component is wrapped in ```<Container>``` and each child will
+ * be wrapped in ```<Col col={12}>{child}</Col>```
+ */
+const Pagination: React.FC<IProps> = ({
+  children,
```

Comment:
I think there is too much responsibility for `children`. From the interface, it is not clear how the children should look like. Do we really need to have children? It could be done a way:
```
SearchComponent
  SearchResulstsComponent
  Pagination
```
so Pagination and SearchResulstsComponent will just visual part and the whole logic inside SearchComponent.
Pagination will be manipulating how many items to show, SearchComponent will filter data (not ReactNodes) and SearchResulstsComponent will render this data accordingly